xbmc/settings/MediaSettings.cpp

```diff
@@ -343,6 +346,15 @@ void CMediaSettings::OnSettingAction(std::shared_ptr<const CSetting> setting)
   }
 }
 
+void CMediaSettings::OnSettingChanged(std::shared_ptr<const CSetting> setting)
+{
+  if (setting == nullptr)
+    return;
+
+  if (setting->GetId() == CSettings::SETTING_VIDEOLIBRARY_SHOWUNWATCHEDPLOTS)
+    CServiceBroker::GetAnnouncementManager()->Announce(ANNOUNCEMENT::VideoLibrary, "xbmc", "OnScanFinished");
```

Comment:
Either change this to `OnUpdate` or add a new type at the following location.
https://github.com/xbmc/xbmc/blob/05611ab739c8f0672ffd0983cdb01d7ed8a5caa8/xbmc/listproviders/DirectoryProvider.cpp#L255-L258

Reply:
Done that, build it and it also works. Decided to use `OnUpdate`

thx